src/transformers/integrations.py

```diff
@@ -305,70 +307,95 @@ def dynamic_modules_import_trainable(*args, **kwargs):
 
 
 def run_hp_search_sigopt(trainer, n_trials: int, direction: str, **kwargs) -> BestRun:
+    import pickle
+    from copy import deepcopy
```

Comment:
Standard library imports should go at the top. Only the one for soft dependencies are delayed.